packages/react-native/ios/HotUpdater/HotUpdater.mm

```diff
@@ -144,34 +139,21 @@ - (BOOL)updateBundle:(NSString *)bundleId zipUrl:(NSURL *)zipUrl {
                 break;
             }
         }
-        
+
         if (filename) {
             NSString *bundlePath = [extractedPath stringByAppendingPathComponent:filename];
             NSLog(@"Setting bundle URL: %@", bundlePath);
-            [self setBundleURL:bundlePath];
-            success = YES;
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [self setBundleURL:bundlePath];
+                if (completion) completion(YES);
+            });
         } else {
             NSLog(@"index.ios.bundle not found.");
-            success = NO;
+            if (completion) completion(NO);
         }
-        
-        dispatch_semaphore_signal(semaphore);
     }];
-    
-    // Add observer for progress updates
-    [downloadTask addObserver:self
-                   forKeyPath:@"countOfBytesReceived"
-                      options:NSKeyValueObservingOptionNew
-                      context:nil];
-    [downloadTask addObserver:self
-                   forKeyPath:@"countOfBytesExpectedToReceive"
-                      options:NSKeyValueObservingOptionNew
-                      context:nil];
```

Comment:
Why did you remove this? Without this code, the progress will not be updated.

Reply:
https://github.com/gronxb/hot-updater/pull/109
I’m restoring the missing part in this PR

Reply:
@gronxb whoops good catch! Thanks that sounds good 👍🏽